docs/CppInteroperability/Roadmap.md

```diff
@@ -0,0 +1,263 @@
+
+```
+// StatefulObject has object identity and reference semantcs: 
+// it should be constructed with "create" and used via a pointer.
+struct StatefulObject {
+  StatefulObject(const StatefulObject&) = delete;
+  StatefulObject() = delete;
+
+  StatefulObject *create() { return new StatefulObject(); }
+};
+```
+
+This type is not intended to be used directly as the type of a local variable or a `std::vector` element; it is allocated on the heap by the `create` method and passed around as a pointer. This is weakly enforced by the way it hides its constructors, but mostly it's communicated in the documentation and by the overall shape of the API; there is no language feature or programming pattern that expresses the specific semantics of this type. 
+
+In Swift, `StatefulObject` would be explicitly defined as a reference type using a specific language construct (defining it as a `class`). Swift defines clear patterns for naming, generic programming, value categories, error handling, and so on, and codebases are encouraged to use these standard practices. These well-defined programming patterns make using Swift APIs a cohesive experience; an experience that C++ interoperability should stive to maintain for Swift programmers using C++ APIs.
+
+To maintain this cohesive experience, the compiler should map C++ APIs to one of these specific Swift programming patterns. To be able to map *any* C++ APIs to *some* specific Swift pattern, the Swift compiler must have some semantic information about the API. This semantic information may be provided through annotations or heuristics in the compiler. In the above example, the user should annotate `StatefulObject` as a reference type so that the Swift compiler can import it as a Swift `class`. 
+
+The Swift compiler should not import APIs for which it does not have sufficent semantic information. It is not a goal to import every C++ API into Swift, especially without additional, required information to present the API in an idiomatic way that promotes a cohesive Swift expirence. The Swift compiler will have a default path for importing APIs where C++ interoperability makes various assumptions about the API that is being imported. Sometimes, as is the case above with `StatefulObject`, an API will fall outside of this path, and will not be imported by default. Other times, the default assumptions will be incorrect for a given API. In both cases, Swift must provide ways for the user to communicate the correct semantics of an API, overriding the default behavior.
```

Comment:
I feel like this can be integrated into the previous paragraph to make a cohesive narrative.  Maybe you can start by talking about the default rules for importing struct types as value types and how `StatefulObject` doesn't fall into them because of the deleted constructors.  You can then briefly explain why we don't want to heuristically *infer* that `StatefulObject` is a reference type; and then you finish up by saying that the user can annotate it explicitly.